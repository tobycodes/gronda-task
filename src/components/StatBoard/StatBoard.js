import React, { useReducer } from "react";

import StatCard from "../StatCard/StatCard";
import Filter from "../Filter/Filter";

import colors from "../../constants/colors";
import { filterReducer } from "../../helpers/reducers";

const StatBoard = () => {
	const [statData, updateStatData] = useReducer(filterReducer, {});
	const { active_source, weekly_active, nps } = statData;

	return (
		<>
			<Filter filterStatData={updateStatData} />
			<div id="stat-board" className="stat-board">
				<StatCard
					statTitle="Active sourcing"
					currentPeriod={active_source?.current_period}
					lastPeriod={active_source?.last_period}
					bgColors={colors[0]}
				/>
				<StatCard
					statTitle="Weekly active"
					currentPeriod={weekly_active?.current_period}
					lastPeriod={weekly_active?.last_period}
					bgColors={colors[1]}
				/>
				<StatCard
					statTitle="NPS"
					currentPeriod={nps?.current_period}
					lastPeriod={nps?.last_period}
					bgColors={colors[2]}
				/>
			</div>
		</>
	);
};

export default StatBoard;
