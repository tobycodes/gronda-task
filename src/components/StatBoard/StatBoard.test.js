import React from "react";
import { shallow } from "enzyme";

import StatBoard from "./StatBoard";

describe("<StatBoard />", () => {
	it("renders a StatBoard component", () => {
		expect(shallow(<StatBoard />)).toMatchSnapshot();
	});
});
