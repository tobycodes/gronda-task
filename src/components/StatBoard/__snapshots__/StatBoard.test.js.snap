// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatBoard /> renders a StatBoard component 1`] = `ShallowWrapper {}`;
